src/status_im2/contexts/quo_preview/wallet/network_routing.cljs

```diff
@@ -0,0 +1,93 @@
+                            :network-name :optimism}
+                           {:amount       initial-amount
+                            :max-amount   (descriptor-state :max-amount-4)
+                            :network-name :polygon}])
+        requesting-data? (reagent/atom false)]
+    (fn []
+      (let [asked-networks (vec (take number-networks @networks))
+            on-success-fn  (fn [new-network-amounts]
+                             (reset! requesting-data? false)
+                             (swap! networks
+                               #(map (fn [network new-amount]
+                                       (assoc network :amount new-amount))
+                                     %
+                                     new-network-amounts)))]
+        [rn/view
+         [quo/network-routing
+          {:total-amount       total-amount
+           :networks           asked-networks
+           :requesting-data?   @requesting-data?
+           :on-amount-selected (fn [new-amount selected-idx]
+                                 (reset! requesting-data? true)
+                                 (fake-call-to-get-amounts
+                                  {:new-amount     new-amount
+                                   :fixed-index    selected-idx
+                                   :current-values (mapv :amount asked-networks)
+                                   :on-success     on-success-fn}))}]
+         (reduce (fn [acc {:keys [amount max-amount color]}]
+                   (conj acc
+                         [rn/view
+                          {:style {:flex-direction  :row
+                                   :margin-vertical 12}}
+                          [rn/view
+                           {:style {:background-color color
+                                    :width            24
+                                    :height           24
+                                    :margin-right     12}}]
+                          [quo/text
+                           "Max limit: " max-amount " Amount: " (subs (str amount) 0 6)]]))
+                 [rn/view {:style {:margin-vertical 12}}
+                  [quo/text "Total amount: " (reduce + (map :amount asked-networks))]]
+                 asked-networks)]))))
+
+(defn preview
+  []
+  (let [descriptor-state (reagent/atom {:total-amount    400
+                                        :number-networks 4
+                                        :max-amount-0    350
+                                        :max-amount-1    350
+                                        :max-amount-2    300
+                                        :max-amount-3    250
+                                        :max-amount-4    200})]
+    (fn []
+      [preview/preview-container {:state descriptor-state :descriptor descriptor}
+       [rn/view {:style {:flex 1 :margin-vertical 28}}
+        ^{:key (str "preview--network-routing-" (:number-networks @descriptor-state))}
```

Comment:
Isn't it a smell that we need to force a React key so the component can be remounted when the number of networks change? This seems to be a sign that the `preview-internal` component is not properly re-rendering based on the props. This seems to be a workaround.

Reply:
Yeah perhaps we need a method in the preview container to reset the page? I had a similar issue in the slider button component where the component initialise with some values and so it needs a hard reset to preview correctly with changes to size prop

Reply:
It's a bug if we assume that any React component correctly implemented should be reflecting the props passed to it. The hard reset should never be necessary in the previews since the component should respect the descriptor values. The other component you mention probably has a bug somewhere too.

1. Could be a bug in our custom code to generate previews, starting at `preview/preview-container`. Unlikely, since most components just work.
2. Could be a bug in the custom code in the preview namespace of the component. This one in this PR is non-trivial, so the bug could be from this quo preview namespace.
3. Most likely a bug in the component itself.

My concern is that the real screens will likely need to do a similar hack. As I mentioned in another comment https://github.com/status-im/status-mobile/pull/17457#discussion_r1344298159, the component as is already doesn't re-render correctly when `networks` change, so I imagine the issue is on option `3`.

Reply:
Explained in the comment below :)